linetools/analysis/voigt.py

```diff
@@ -0,0 +1,203 @@
+      Observed wavelengths
+    line: AbsLine, List of Absline, or array of parameters
+    skip_wveval: bool, optional [False]
+      Skip wavelength check
+      If False, the median dwave value is calculated and compared against
+      1/10 of the b-value of the input line(s).  
+      If necessary, the wavelenght array is rebinned for the calculation
+      and the final array is rebinned to the original.
+    fwhm: float, optional [None]
+      FWHM for Gaussian smoothing (pixels)
+    flg_ret : int (1)  Byte-wise Flag for return
+      1: vmodel [XSpectrum1D]
+      2: tau
+      4: Absorbed flux [np.ndarray]
+    ToDo:
+        1.  May need to more finely sample the wavelength array
+    '''
+    # Wavelength input
+    if not isinstance(iwave,Quantity):  # Standard wavelength array
+        raise ValueError('voigt_model: Unknown spectrum input')
+
+    # Demand that the evaluation delta wavelength is < 1/10 of the b-value
+    flg_rebin = 0
+    if not skip_wveval:
+        if isinstance(line,AbsLine):  # Single line as a AbsLine Class
+            minb = line.attrib['b']
+        else:
+            minb = np.min(np.array(
+                [iline.attrib['b'].value for iline in line]))*u.km/u.s
+        # Calculate dwave
+        dwave = np.median(np.abs(iwave-np.roll(iwave,1)))
+        medwave = np.median(iwave)
+        warnings.warn('Using a sub-grid wavelength array because the input array is too coarse.')
+        warnings.warn('Will return values rebinned to the input array.')
+        if const.c.to('km/s')*dwave/medwave > minb/10.:
+            wmin = np.min(iwave.to('AA').value)
+            wmax = np.max(iwave.to('AA').value)
+            nsub = int(np.round( (np.log10(wmax)- np.log10(wmin)) / 1.449E-6)) + 1
+            wave = 10.**(np.log10(wmin) + np.arange(nsub)*1.449E-6) * u.AA
+            flg_rebin = 1
+        else:
+            wave = iwave
+    else: # No evaluation 
+        wave = iwave
+
+    # Line input
+    if isinstance(line,AbsLine):  # Single line as a AbsLine Class
+        pass
+        #par = [0*i for i in range(6)] # Dummy list
+        #par[0] = line.attrib['N'] # logN; Won't have units
+        #par[1] = line.attrib['z']
+        #par[2] = line.attrib['b'] # Should have units
+        #par[3] = line.wrest # Should have units
+        #par[4] = line.data['f']
+        #par[5] = line.data['gamma'] # Should have units
+    elif isinstance(line,list):
+        if isinstance(line[0],AbsLine):  # List of Abs_Line
+            tau = np.zeros(len(wave))
+            for iline in line:
+                if debug:
+                    print(iline)
+                tau += voigt_model(wave, iline, fwhm=0., 
+                    flg_ret=2, debug=debug, skip_wveval=True) 
+        else:
+            raise ValueError('voigt: Unknown type for voigt line in your list')
+    else: 
+        raise ValueError('voigt: Unknown type for voigt line')
+
+    # tau
+    if 'tau' not in locals(): # Deals with multiple line input
```

Comment:
Move this block (150-169) up inside the `if` block starting at line 126, instead of testing here for tau in locals().